pkg/deploy/generator/scripts/gatewayVMSS.sh

```diff
@@ -316,7 +316,20 @@ echo "Download \$COMPONENT credentials"
 
 TEMP_DIR=\$(mktemp -d)
 export AZURE_CONFIG_DIR=\$(mktemp -d)
-az login -i --allow-no-subscriptions
+
+echo "Logging into Azure..."
+RETRIES=3
+while [ "\$RETRIES" -gt 0 ]; do
```

Comment:
I think this quoting will result in a bash error: `$RETRIES: integer expression expected`, since we are comparing the literal string (with dollar sign) `$RETRIES` to integer `0`